backend-shared/hack/ci-check/main.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+)
+
+const (
+	DbSchemaRelativeFileLocation         = "../db-schema.sql"
+	DbFieldConstantsRelativeFileLocation = "./config/db/db_field_constants.go"
+)
+
+func main() {
+
+	fieldToSize := parseDbSchema()
+	fieldConstantToSize := parseDbConstants()
+	checkIfSchemaInSyncWithConstants(fieldConstantToSize, fieldToSize)
+
+}
+
+func checkIfSchemaInSyncWithConstants(fieldConstantToSize map[string]string, fieldToSize map[string]string) {
+
+	for fieldName, fieldSize := range fieldConstantToSize {
+		if _, ok := fieldToSize[fieldName]; !ok {
+			err := fmt.Errorf("field %s not present in the schema", fieldName)
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		if fieldToSize[fieldName] != fieldSize {
+			err := fmt.Errorf("sizes for the field %s are not in sync", fieldName)
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+	}
+
+	for fieldName, fieldSize := range fieldToSize {
+
+		if _, ok := fieldConstantToSize[fieldName]; !ok {
+
+			err := fmt.Errorf("field %s not present as a constant in the db_field_constants.go file", fieldName)
+			fmt.Println(err)
+			os.Exit(1)
+		}
+
+		if fieldConstantToSize[fieldName] != fieldSize {
+
+			err := fmt.Errorf("sizes for the field %s are not in sync", fieldName)
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	}
+}
+
+func parseDbSchema() map[string]string {
+
+	fieldToSize := make(map[string]string)
+	dbSchemaFile, err := os.ReadFile(DbSchemaRelativeFileLocation)
+
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	dbSchema := strings.Split(string(dbSchemaFile), "\n")
+
+	i := 0
+	for i < len(dbSchema) {
```

Comment:
@drpaneas in this case since there is a nested loop which uses the same index variable, it makes it easier to use the index and manually increment it instead of using range. Can we keep it this way or does it need to be changed?

Reply:
Okayy. Thank you!